package.json

```diff
@@ -32,7 +32,9 @@
     "lint:fix": "eslint --fix",
     "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc",
     "deploy-ftp": "export $(cat .env | xargs) && node deploy-ftp.js",
-    "build:deploy-ftp": "git checkout dev && git pull && npm run build && npm run deploy-ftp"
+    "build:deploy-ftp": "git checkout dev && git pull && npm run build && npm run deploy-ftp",
+    "precommit": "npm run lint:fix && git add . && npm run format && git add .",
+    "configure-husky": "npx husky install && npx husky add .husky/pre-commit 'npx --no-install lint-staged'"
```

Comment:
Só para ver se percebo: se eu trabalhar num branch, e fizer um commit que tem lint errors. Vou ter de correr algum commando adicional para corrigir os errors, ou os erros sao corrigidos automaticamente quando faço um commit? @juliamendesc

Reply:
Oi! Não, o precommit é apenas pros arquivos que já estão lá (não fiz e commitei a correção porque saía do escopo da task).

O que acontecerá é que todas as vezes que for fazer um commit, ele vai verificar se tem erros de lint e formatar. Se tiver algum erro, ele não te deixa fazer o commit. Esse vai ser o default.

Se preferir, eu faço já a formatação dos arquivos da branch e retiro o script precommit.

Reply:
Ah já percebi! Interessante :) Eu no trabalho tenho `pre-push` em vez de `pre-commit`. Vou deixar espaço aqui para outros mentores darem a sua opinião e reverem cc @toothlesspanda @gustavocr19-developer 

Sobre a formatação dos arquivos todos, acho que podias fazer noutra PR com apenas essas modificacoes, para ser mais fácil rever como follow up. Se quiseres força nisso @juliamendesc

Reply:
Boa! Eu usei o husky num outro projeto e amei, mas como uso o prettier no onSave, não tinha feito o format dentro dele. Daí a implementação aqui foi diferente e aceito todo feedback <3

Reply:
Vou olhar suas alterações :)

Reply:
@juliamendesc pra ver se entendi, o husky vai ser acionado no pre-push ? 🤔
Porque clonei seu código na minha máquina, instalei as dependências, alterei alguns arquivos(deixei com erros de lint), fiz o commit e o push. Nesse processo não consegui ver o script ser executado. Deveria ser esse o passo a passo ?

Outra coisa, percebi que o "lint:fix": "eslint --fix" na linha 32 não está a funcionar quando é executado separadamente("npm run lint:fix"). O correto seria "eslint . --fix" (como fez na linha 73).  Disse isso, porque está a usar esse comando no "pre-commit".

Se precisar de ajuda estou a disposição 🙂

Reply:
Oi @gustavocr19-developer! 😄 Bora lá:

Primeiro: o lint --fix não corrige tudo e isso não é uma falha do nosso lado, mas do eslint mesmo. Você pode ver [aqui](https://eslint.org/docs/latest/rules/) quais regras têm suporte para correções automáticas (estão com a ferramenta em highlight). Portanto, o que for possível corrigir, ele vai corrigir. O que não puder e estiver classificano no .eslintrc como erro, vai jogar erro e abortar o script. Se tiver apenas classificado no .estlintrc como warning, ele continua o script e mostra os warnings.

A forma como coloquei realiza duas coisas em separado: 
1. Se você correr manualmente o npm run precommit, ele vai correr o lint e o prettier em todos os arquivos. Essa é uma solução provisória que propus para que vocês pudessem ver em quais arquivos há erros de lint que não são corrigíveis automaticamente e, portanto, vocês precisam dar alguma atenção (a maioria é de unused vars).

2. Quando estiverem trabalhando e fizerem stage dos arquivos, ele vai correr a mesma coisa nos arquivos em stage. 

Pelo que entendi, foi esse segundo que não funcionou com você, correto? Eu dei update no código porque identifiquei uns errinhos mesmo no script do pre-commit. Ele agora está só assim para os arquivos que vão a stage. 
![image](https://github.com/As-Raparigas-do-Codigo/ardc-website/assets/57814821/1a4039ca-74f5-494a-83c4-6d70f581ae7c)

Aqui um gif de como ele deve funcionar no commit:
![lintOnCommit](https://github.com/As-Raparigas-do-Codigo/ardc-website/assets/57814821/16f6b6e4-6abd-48e1-83da-98b70a6f1719)

Sobre a linha 32, eu tentei dos dois jeitos e funcionou normal comigo:
Com o comando que você disse:
![image](https://github.com/As-Raparigas-do-Codigo/ardc-website/assets/57814821/d5a07135-3e8d-47cb-9aff-e7f50b76255a)

Com o comando que estava:
![image](https://github.com/As-Raparigas-do-Codigo/ardc-website/assets/57814821/63f4eae5-ac06-40a1-a7aa-818e8f1db44f)

Mas isso sem problema, coloco o que vc preferir e tiro o lint:fix porque acho que vai ficar obsoleto 🙂

Aqui um gif do lint funcionando e corrigindo a regra de no-vars (ele muda var por let):
![lint](https://github.com/As-Raparigas-do-Codigo/ardc-website/assets/57814821/fdf70ad7-b3d6-414a-b8d3-a7d844091b39)

Reply:
Em follow-up, se preferir, to disponível pra marcar uma call pra me fazer entender melhor se preferirem. Tento ser o mais clara possível, mas acho que às vezes só falando mesmo XD

Reply:
@juliamendesc Parabéns pelo excelente trabalho e dedicação ❤️

Ficou tudo muito claro com a nova explicação e exemplos. 🙂
Sobre o "lint --fix" pode deixar como está, sem problemas.